pkg/util/typeutil/errors.go

```diff
@@ -0,0 +1,70 @@
+package typeutil
```

Comment:
In my own code, I've been trying to reduce the use of `util`packages, because they kind of become a dumping ground of things that can be put into the 1 or 2 packages they end up only serving. Do you think this code would be better off in the `pkg/container/types` package instead? None of it is importing the `v1` code, so it shouldn't be an import cycle.

Reply:
It's going to end up being shared with the validation code for workflow files. However, if there's a better general name you'd like to suggest, I'd be happy to use it!

Reply:
I don't have any suggestions at the moment. Could the validation code just pull in the `pkg/container/types` package? It just seems to make more sense to scope the code under there instead of some adhoc utility package in the root. This is just approaching it from an SDK prospective where someone will be looking at how to use certain sections of the code and bouncing to some other adhoc package might be confusing.

Reply:
I could put the `types` package at the top level, I suppose. Right now everything under `pkg/container` is specific to the container generator, and I'm not sure importing from outside there is going to make it any cleaner from an outsider's perspective... :/